rmw_zenoh_cpp/src/rmw_init_options.cpp

```diff
@@ -76,9 +77,8 @@ rmw_init_options_copy(const rmw_init_options_t * src, rmw_init_options_t * dst)
   RCUTILS_CHECK_ALLOCATOR(&allocator, return RMW_RET_INVALID_ARGUMENT);
 
   rmw_init_options_t tmp;
-  tmp.instance_id = src->instance_id;
+  memcpy(&tmp, src, sizeof(rmw_init_options_t));
```

Comment:
**Suggestion:** Ensure that `memcpy` is safe to use with `rmw_init_options_t` by checking if it contains any non-trivially copyable types. [possible issue, importance: 9]
```suggestion
  if constexpr (std::is_trivially_copyable<rmw_init_options_t>::value) {
      memcpy(&tmp, src, sizeof(rmw_init_options_t));
  } else {
      // Implement or use a copy constructor
  }
```